pdf/creator/table.go

```diff
@@ -476,12 +519,59 @@ func (cell *TableCell) SetVerticalAlignment(valign CellVerticalAlignment) {
 // SetBorder sets the cell's border style.
 func (cell *TableCell) SetBorder(style CellBorderStyle, width float64) {
 	cell.borderStyle = style
-	cell.borderWidth = width
+	if style == CellBorderStyleBox {
+		cell.borderWidthLeft = width
+		cell.borderWidthBottom = width
+		cell.borderWidthRight = width
+		cell.borderWidthTop = width
+	} else if style == CellBorderStyleBoxLeft {
+		cell.borderWidthLeft = width
+	} else if style == CellBorderStyleBoxBottom {
+		cell.borderWidthBottom = width
+	} else if style == CellBorderStyleBoxRight {
+		cell.borderWidthRight = width
+	} else if style == CellBorderStyleBoxTop {
+		cell.borderWidthTop = width
+	}
 }
 
 // SetBorderColor sets the cell's border color.
 func (cell *TableCell) SetBorderColor(col Color) {
-	cell.borderColor = model.NewPdfColorDeviceRGB(col.ToRGB())
+	cell.borderColorLeft = model.NewPdfColorDeviceRGB(col.ToRGB())
+	cell.borderColorBottom = model.NewPdfColorDeviceRGB(col.ToRGB())
+	cell.borderColorRight = model.NewPdfColorDeviceRGB(col.ToRGB())
+	cell.borderColorTop = model.NewPdfColorDeviceRGB(col.ToRGB())
+}
+
+// SetBorder sets the cell's border style.
+func (cell *TableCell) SetBorderColorAndWidthByStyle(style CellBorderStyle, col Color, width float64) {
```

Comment:
Very long too informative function name. Can we improve?   It is also a bit confusing that we need to specify the style again here.  It appeared that the style was already applied.  It might be better if we can refer to a specific side rather than style.  Maybe SetBorderBySide to set specific side properties rather than border properties for entire cell?

Reply:
I will try something else as per your recommendation.